python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -489,6 +490,34 @@ def compute_latest_time_window_subset(
         else:
             check.failed(f"Unsupported partitions_def: {partitions_def}")
 
+    def compute_downstream_subsets(
+        self, input_subsets: Sequence[EntitySubset]
+    ) -> Sequence[EntitySubset]:
+        """Computes all downstream subsets from the input subsets, inclusive of the original input.
+        This operates recursively, meaning child subsets of child subsets are included.
```

Comment:
suggestion below seems particularly important perf-wise for self-dependant assets. Imagine a daily partitioned asset and you're only backfilling a small range of days from a long time ago. If you're intersecting with the target subset you won't have to keep traversing downwards at all most likely. Without it, you need to keep traversing over and over again until you reach the present day.